symbolic-il2cpp/src/line_mapping/from_object.rs

```diff
@@ -115,16 +116,19 @@ impl<'data> Iterator for SourceInfos<'data> {
     type Item = SourceInfo<'data>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        for (cpp_line, cpp_src_line) in &mut self.lines {
+        while let Some((_, cpp_src_line)) = self.lines.next() {
             match parse_line(cpp_src_line) {
+                None => continue,
                 Some((cs_file, cs_line)) => {
+                    let next_non_comment_line_nr = self.lines.find_map(|(nr, src_line)| {
```

Comment:
Is `find_map` defined as being "short circuiting"? It would be bad if it exhausted the whole iterator.

Reply:
Yes, it stops as soon as it finds the first `Some` return value. It's equivalent to `filter_map(…).next()`.